### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by thirionlogan**
Implement authentication endpoints

### CODE REVIEW

File: server/app/app.js [hunk 1]
```diff
@@ -82,7 +83,29 @@ app.get('/', (req, res) => {
...
+    .then(() => {
+      res.sendStatus(201);
+    })
+    .catch((err) => {
+      res.sendStatus(422);
+    });
+});
+
+app.post('/api/login', async (req, res) => {
+  authenticateLogin(req.body)
```
Comment (thirionlogan) [inline server/app/app.js:97]: returns the user if the email/password is correct

File: server/app/app.js [hunk 2]
```diff
@@ -82,7 +83,29 @@ app.get('/', (req, res) => {
...
+    .then((user) => {
+      req.session.user = user;
```
Comment (thirionlogan) [inline server/app/app.js:99]: sets the found user to the sessions so that the user is persisted

File: server/app/app.js [hunk 3]
```diff
@@ -82,7 +83,29 @@ app.get('/', (req, res) => {
...
+      res.status(200).send(user);
+    })
+    .catch(() => {
+      res.sendStatus(401);
+    });
+});
+
+app.post('/api/logout', async (req, res) => {
+  req.session.user = undefined;
```
Comment (thirionlogan) [inline server/app/app.js:108]: erase the user from the session on log out

File: server/data/seeds/001_seed_user.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    .del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('user').insert([
+        {
+          firstName: 'John',
+          lastName: 'Doe',
+          email: 'johndoe@email.com',
+          password:
+            '$2a$10$YSPUaf2rbvPwyxROi7zOaeEQQUM6G1m.1K3dkoJzrGYlAIlilnEre',
```
Comment (thirionlogan) [inline server/data/seeds/001_seed_user.js:13]: salted and hashed from "password"

File: server/services/userService.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    email,
+    password,
+    confirmPassword,
+  });
+  if (!validationResult.valid) throw new Error(validationResult.message);
+  return new User({
+    email,
+    firstName,
+    lastName,
+    password: bcrypt.hashSync(password, bcrypt.genSaltSync(10)),
```
Comment (thirionlogan) [inline server/services/userService.js:30]: salt and hash password before it is stored in the database

**Review by knieszj — APPROVED** [review]

**Review by adder206 — APPROVED** [review]
Reviewed on live session.  Approved.

**Pull request merged**